src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -233,18 +239,31 @@ def _process_message(wrapper: Dict) -> None:
                         monitor_environment.last_checkin
                     )
 
-                check_in = MonitorCheckIn.objects.create(
-                    project_id=project_id,
-                    monitor=monitor,
-                    monitor_environment=monitor_environment,
-                    guid=params["check_in_id"],
-                    duration=duration,
-                    status=status,
-                    date_added=date_added,
-                    date_updated=start_time,
-                    expected_time=expected_time,
-                    monitor_config=monitor.get_validated_config(),
+                check_in_guid = params["check_in_id"]
+                lock = locks.get(
+                    f"checkin-creation:{check_in_guid}", duration=2, name="checkin_creation"
                 )
+                try:
+                    with lock.acquire():
+                        check_in = MonitorCheckIn.objects.create(
```

Comment:
The second checkin creation is still going to fail here after it acquires the lock, should we be using `get_or_create`? Is the second checkin meant to update fields if there's a race or do we just want to drop?

Reply:
good call. will work out the right flow and update